src/internal/middleware/auth/interceptor.go

```diff
@@ -279,6 +280,13 @@ type Interceptor struct {
 	getAuthServer func() authserver.APIServer
 }
 
+func peerName(ctx context.Context) string {
```

Comment:
Seems a bit surprising that it is called `peerName` but may not actually return a peer name. Looking at how it’s used, it doesn’t make sense to complicate its signature; maybe rename to `peerNameOrUnknown`, `peerString` or something like that?

Or perhaps I am being too nitpicky …